crates/analyzer/src/constants.rs

```diff
@@ -1 +1,4 @@
+pub const EMITTABLE_TRAIT_NAME: &str = "Emittable";
+pub const EMIT_FN_NAME: &str = "emit";
+pub const INDEXED: &str = "indexed";
```

Comment:
I could move these to a place closer to their usage. My reasoning for putting them here is that they are simple strings and I dislike using magic strings inline because it is easy to mistype a string without the compiler noticing (e.g. "inexed" but you can not mistype `INDEXED` without the compiler noticing it.

Reply:
Why should `INDEXED` or `EMITTABLE` be used from several modules? I think exposing constants in an arbitrary way is not good especially if you know these constants are used as a stopgap.  How long do you think these constants will remain?